pandas/plotting/_matplotlib/core.py

```diff
@@ -332,11 +332,14 @@ def _compute_plot_data(self):
         # GH16953, _convert is needed as fallback, for ``Series``
         # with ``dtype == object``
         data = data._convert(datetime=True, timedelta=True)
-        numeric_data = data.select_dtypes(include=[np.number,
-                                                   "datetime",
-                                                   "datetimetz",
-                                                   "timedelta"])
-
+        # GH26173, don't filter out numeric extension types
+        # Numeric categorical data gets caught by including np.number
+        # in select_dtypes
+        numeric_dtypes = [dtype for dtype in set(data.dtypes)
+                          if hasattr(dtype, "_is_numeric") and
+                          dtype._is_numeric and
```

Comment:
rather than this i would fix select_dtypes to understand the is_numeric flag of the EA dtype; it’s possible that actually need something more because for example datetimes are not numeric but are allowed for plotting

Reply:
Agreed that this should be addressed in select_dtypes rather than here.